libs/utils/state_wrapper.go

```diff
@@ -0,0 +1,34 @@
+package utils
+
+import "sync/atomic"
+
+type state uint32
+
+const (
+	Stopped state = iota
+	Runnnig
+)
+
+// StateWrapper provides an interface to change an check service state
+type StateWrapper interface {
```

Comment:
StateWrapper is a wrong name for the interface, IMO. Also, do we really need this interface? Do you expect multiple implementations of it?  What about a simple Service struct that all our services embed through composition?

Reply:
removed